persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -411,46 +411,92 @@ public static void inferRelationDetails(Module entityModule) {
         Map<String, Entity> entityMap = entityModule.getEntityMap();
         for (Entity entity : entityMap.values()) {
             List<EntityField> fields = entity.getFields();
-            fields.stream().filter(field -> field.getRelation() != null).filter(field -> field.getRelation().isOwner())
+            fields.stream().filter(field -> entityMap.get(field.getFieldType()) != null)
                     .forEach(field -> {
                         String fieldType = field.getFieldType();
-                        field.getRelation().setRelationType(field.isArrayType() ?
-                                Relation.RelationType.MANY : Relation.RelationType.ONE);
                         Entity assocEntity = entityMap.get(fieldType);
```

Comment:
Seems like we process all entities together, wouldn't this run the logic twice? For instance if A->B, 
This logic first runs for A's field, then B's field.

Reply:
No, we only process when the Relation is null and set the relation for the assoc entity field. At the time when processing the assoc entity, it has the relation object. So it will not run the logic twice.

Reply:
Ack